core/workspace_svg.js

```diff
@@ -1666,6 +1679,30 @@ Blockly.WorkspaceSvg.prototype.getButtonCallback = function(key) {
   return this.flyoutButtonCallbacks_[key];
 };
 
+/**
+ * Register a callback function associated with a given key, for populating
+ * custom toolbox categories in this workspace.  See the variable and procedure
+ * categories as an example.
+ * @param {string} key The name to use to look up this function.
+ * @param {function(!Blockly.Workspace):!Array<!Element>} func The function to
+ *     call when the given toolbox category is opened.
+ */
+Blockly.WorkspaceSvg.prototype.registerToolboxCategoryCallback = function(key,
```

Comment:
Why workspace?  My understand is that Workspace has no UI so there'd be no way to open a toolbox category and have this fire in a headless environment.